### DISCUSSION TITLE
[Listbox] - Add hidden inputs that use the same name property as the root Listbox

### DISCUSSION THREAD

**Discussion by cevr** [tailwindlabs/headlessui#1448] [Ideas]
Currently the native form support for the `multiple` attribute in a listbox adds hidden inputs with names like `name[valueIndex]`. This makes it a bit awkward to consume when trying to use the native [FormData.getAll(name)](https://developer.mozilla.org/en-US/docs/Web/API/FormData/getAll).

I do not know the use-case for the current implementation, but would it be possible to add another hidden input per value that uses the name given to the root Listbox?

**Comment by gafemoyano** [comment]
Hi, did you eventually figure out a way to retrieve `multiple` form values using FormData?

**Reply by cevr** [reply]
Yes, I created hidden inputs with the name given to the root inbox

**Comment by jonathansgardner** [comment]
I used a helper function to parse the values stored in headlessui's implemenetation of hidden inputs on a listbox or combobox component that is marked multiple.

```
// returns an array containing all non-null values for inputs with a name attribute of fieldName[0], fieldValue[1], etc.
export default function getIndexedFormDataArray(
  formData: FormData,
  fieldName: string,
): FormDataEntryValue[] {
  const array = [];
  let i = 0;

  while (formData.has(`${fieldName}[${i}]`)) {
    const value = formData.get(`${fieldName}[${i}]`);

    if (value !== null) {
      array.push(value);
    }

    i++;
  }

  return array;
}
```
